src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -167,6 +176,26 @@ public static bool IsWindowsConsumption(this IEnvironment environment)
             return string.Equals(value, ScriptConstants.DynamicSku, StringComparison.OrdinalIgnoreCase);
         }
 
+        /// <summary>
+        /// Returns true if the app is running on Virtual Machine Scale Sets (VMSS)
+        /// </summary>
+        public static bool IsVMSS(this IEnvironment environment)
+        {
+            string value = environment.GetEnvironmentVariable(EnvironmentSettingNames.RoleInstanceId);
```

Comment:
@safihamid Please review this VMSS check

Reply:
looks good.

---
In reply to: [413449441](https://github.com/Azure/azure-functions-host/pull/5897#discussion_r413449441) [](ancestors = 413449441)